### PR TITLE
English Overhaul of Documentation

### PR DESCRIPTION
**Pull request by EvaBS**
- Parts 00 - 30 have minor changes
- Part 50 needs Reviewer's attention because I changed the order of the Lens docs
- Parts 40-100 received an english text overhaul
- Changed the document order number for the www structure according to the filenames (2 were wrong)
- Removed entries from todo file which were already done

### CODE REVIEW

File: www/src/pages/docs/50_StoreMapping.md [hunk 1]
```diff
@@ -12,63 +12,72 @@ eleventyNavigation:
...
 
-Like the known `map`-function from collections, where some source type `T` gets transformed to some other type `R`
-inside an expression, we can also *map* a store in order to change its source type to some other, better fitting type.
-There is one big difference between the classical `map`-function and the store's mapping functions: A store needs
-not only a function from `T -> R` (getter) but also from `R -> T` (setter) as a store manages changes!
+Remember the `map`-function from collections where a source type `T` is transformed to another type `R`
+inside of an expression? We can also *map* a store in order to change its source type to a more suitable type.
+However, there is one big difference between the classical `map`-function and the store's mapping functions: A store needs
+both a getter-function from `T -> R` and a setter-function from `R -> T` to manage changes, while the collections functions
+only requires the getter.
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:27]: In collections `map` that is not a "getter"; so I would drop the latter remark.

File: www/src/pages/docs/100_Web Components.md [hunk 2]
```diff
@@ -79,18 +79,17 @@ object WeatherCard : WebComponent<HTMLDivElement>() {
         }
 }
 ```
-We also add an observed attribute named `city` which then can be set by the user of our component.
-Next, register your component:
+Also add an observed attribute named `city` to be set by the user of this component. Next, register the component and add the observed `city` to the registration.
```
Comment (Lysander) [inline www/src/pages/docs/100_Web Components.md:82]: Respect the maximum line length please and add some line break. (That might be needed in other places as well!)

File: www/src/pages/docs/50_StoreMapping.md [hunk 3]
```diff
@@ -12,63 +12,72 @@ eleventyNavigation:
...
 
-There is a universal concept in computer science for such a functionality called *lens*. You might have a look at
-the [excellent documentation on lenses](https://arrow-kt.io/learn/immutable-data/lens/) from the 
-[arrow-project](https://arrow-kt.io).
+If you are unfamiliar with the universal concept for the functionality called *lens* in computer science, take
+a look at the [excellent documentation on lenses](https://arrow-kt.io/learn/immutable-data/lens/) from the 
+[arrow-project](https://arrow-kt.io) before reading on.
 
-fritz2 also offers the method `lensOf()` for a short-and-sweet-experience, which accepts a getter- and 
+A `Lens` is basically a way to describe the relation between an outer and inner entity in a structure. It focuses on the
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:35]: I prefer the old structure - this is really just the teasing teaser, not the 100% in detail explanation.
In the upcoming dedicated section it is the right place to describe the details - at this place, we should omit them to focus on the big picture.

File: www/src/pages/docs/50_StoreMapping.md [hunk 4]
```diff
@@ -12,63 +12,72 @@ eleventyNavigation:
...
 val hugo: Person = nameLens.set(fritz2, "hugo") // hugo = Person(1, "hugo")
 ```
-As you can see, there is no *magic*; just plain old function calling.
+As you can see, there is no magic, just plain old function calling.
 
-Let us take a step back and explore, how this concept of lenses can be used to map one store to another.
+Let's take a step back and explore how the concept of lenses can be used to map one store to another.
 
-### Mapping a Store
+### Mapping Complex Stores Using Lenses
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:61]: Why complex? It emphasizes some property, which is not important here!?

File: www/src/pages/docs/50_StoreMapping.md [hunk 5]
```diff
@@ -311,8 +308,10 @@ val person = Person(Address("Lerchenweg"))
 streetOfPerson.get(person) // -> "Lerchenweg"
 streetOfPerson.set("Rosenstraße") // Person(address = Address("Rosenstraße"))
 ```
+This works, but the syntax is quite cumbersome - especially for deeper nested models.
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:311]: This syntax is not meant! The `+` Operator is cumbersome for deeper nested models ;-)
So the old appearence is valid.

File: www/src/pages/docs/50_StoreMapping.md [hunk 6]
```diff
@@ -358,15 +355,9 @@ val ageLens: Lens<Person, Int> = Person.age() // cannot be used in tag attribute
 val ageLensAsString: Lens<Person, String> = Person.age().asString() // now it is usable
 ```
 
-fritz2 also provides a special `lensOf()` function for creating a `Lens<P, String>` for special types that are not
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:361]: Why omit the code here? Repeitition is useful here - the documentation could be read without reading the `Overview` sections! Getting the signature here is imho a fine thing.

File: www/src/pages/docs/50_StoreMapping.md [hunk 7]
```diff
@@ -378,53 +369,53 @@ object Formats {
...
-| `lensForElement(index: Int): Lens<List, T>`                               | Select one element from a list by index. Useful for value objects                               |
-| `lensForElement(key: K): Lens<Map<K, V>, V>`                              | Select one element from a map by some key.                                                      |
+| `lensForElement(element: T, idProvider: IdProvider<T, I>): Lens<List, T>` | Select one element from a list of entities, therefore a stable Id is needed.                    |
+| `lensForElement(index: Int): Lens<List, T>`                               | Select one element from a list by index. Useful for value objects.                              |
+| `lensForElement(key: K): Lens<Map<K, V>, V>`                              | Select one element from a map by key.                                                           |
 
 ## Advanced Topics
 
-### Reactive Rendering of Lists of Entities with automatically Mapped Element Store
+### Reactive Rendering of Entity-Lists With Auto-Mapped Store
```
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:408]: The "element" should be kept! This is the clue of this special feature after all.

File: www/src/pages/docs/50_StoreMapping.md (hunk 1 above)
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:27]: In collections `map` that is not a "getter"; so I would drop the latter remark.

File: www/src/pages/docs/100_Web Components.md (hunk 2 above)
Comment (Lysander) [inline www/src/pages/docs/100_Web Components.md:82]: Respect the maximum line length please and add some line break. (That might be needed in other places as well!)

File: www/src/pages/docs/50_StoreMapping.md (hunk 3 above)
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:35]: I prefer the old structure - this is really just the teasing teaser, not the 100% in detail explanation.
In the upcoming dedicated section it is the right place to describe the details - at this place, we should omit them to focus on the big picture.

File: www/src/pages/docs/50_StoreMapping.md (hunk 4 above)
Comment (Lysander) [inline www/src/pages/docs/50_StoreMapping.md:61]: Why complex? It emphasizes some property, which is not important here!?

File: www/src/pages/docs/50_StoreMapping.md (hunk 3 above)
Comment (EvaBS) [reply www/src/pages/docs/50_StoreMapping.md:34]: I dont think this is a detail, it's what helps you understand what a lens even is.

Comment (Lysander) [reply www/src/pages/docs/50_StoreMapping.md:34]: The base concept of the documentation is to present the user an *overview*, what is possible with fritz2. In order to get an impression of store mapping, the current *short* teasing of lenses is sufficient. It is intentionally short and hides detailed information for later explanation.

It is simply not the right way, to move those details up here! It contradicts the "new" approach of the documentation, which clearly separates the "Overview" from the "Essentials".

Comment (EvaBS) [reply www/src/pages/docs/50_StoreMapping.md]: Undone ..

**Review by Lysander — APPROVED** [review]

**Pull request merged**